functionary/prompt_template/base_template.py

```diff
@@ -149,14 +104,16 @@ def get_prompt_from_messages(
         Returns:
             str: the prompt for inference/training
         """
-        messages_clone = self.inject_system_messages_based_on_tools(
-            messages, tools_or_functions
+
+        jinja_template = self._jinja_env.from_string(self.get_chat_template_jinja())
```

Comment:
Actually as I mentioned before, maybe we should store this as a parameter instead of creating a new instance every time because in packing, we need to **handle all the data points ahead**, not sure if this takes time or not